core/designsystem/src/main/kotlin/com/google/samples/apps/nowinandroid/core/designsystem/component/scrollbar/ScrollbarExt.kt

```diff
@@ -102,3 +104,40 @@ fun LazyGridState.scrollbarState(
         },
         reverseLayout = { layoutInfo.reverseLayout },
     )
+
+/**
+ * Remembers a [ScrollbarState] driven by the changes in a [LazyStaggeredGridState]
+ *
+ * @param itemsAvailable the total amount of items available to scroll in the staggered grid.
+ * @param itemIndex a lookup function for index of an item in the staggered grid relative
+ * to [itemsAvailable].
+ */
+@Composable
+fun LazyStaggeredGridState.scrollbarState(
+    itemsAvailable: Int,
+    itemIndex: (LazyStaggeredGridItemInfo) -> Int = LazyStaggeredGridItemInfo::index,
+): ScrollbarState =
+    scrollbarState(
```

Comment:
This looks really complex. Is there any way we could simplify this code?

Reply:
It can be simplified if I copy paste the logic for `LazyListState`, `LazyGridState` and `LazyStaggeredGridState` instead of conforming them all to the same abstraction. Is that preferred?

Reply:
The current code is quite complex to follow. I'd prioritize readability and simplicity over avoiding redundancy.

Reply:
Fixed!